src/sched.rs

```diff
@@ -184,3 +189,13 @@ pub fn unshare(flags: CloneFlags) -> Result<()> {
 
     Ok(())
 }
+
+pub fn setns(fd: RawFd, nstype: CloneFlags) -> Result<()> {
+    let res = unsafe { ffi::setns(fd, nstype) };
+
+    if res != 0 {
+        return Err(Error::Sys(Errno::last()));
+    }
+
+    Ok(())
```

Comment:
You can replace most of that with `from_ffi(res)`.

I've been meaning to send a PR that does this across the codebase. Maybe I'll do that this afternoon!